internal/testrunner/testrunner.go

```diff
@@ -34,10 +30,23 @@ type TestOptions struct {
 	DeferCleanup time.Duration
 }
 
-// RunFunc method defines main run function of a test runner.
-type RunFunc func(options TestOptions) ([]TestResult, error)
+// TestRunner is the interface all test runners must implement.
+type TestRunner interface {
+	// Type returns the test runner's type.
+	Type() TestType
+
+	// String returns the human-friendly name of the test runner.
+	String() string
+
+	// Run executes the test runner.
+	Run(TestOptions) ([]TestResult, error)
+
+	// TearDown cleans up any test runner resources. It must be called
+	// after the test runner has finished executing.
+	TearDown()
```

Comment:
is it fair for `TearDown` method to return an error?

Reply:
Yeah, I had it returning an error at one point but looking at the current implementations of `TearDown`, none of them actually return an error so I took it out.

But, since you're also thinking about it, I'm once again leaning towards allowing `TearDown` to return an error. I will add it back, thanks!

Reply:
Added in 7de930cc20dedf76ae7752210988e9413ff5b71a.